app/assets/javascripts/discourse/app/components/user-menu/notifications-list.js

```diff
@@ -60,27 +69,77 @@ export default class UserMenuNotificationsList extends UserMenuItemsList {
       limit: 30,
       recent: true,
       bump_last_seen_reviewable: true,
-      silent: this.currentUser.enforcedSecondFactor,
     };
 
+    if (this.currentUser.enforcedSecondFactor) {
+      params.silent = true;
+    }
+
     const types = this.filterByTypes;
     if (types?.length > 0) {
       params.filter_by_types = types.join(",");
       params.silent = true;
     }
-    const collection = await this.store
-      .findStale("notification", params)
-      .refresh();
-    const notifications = collection.content;
-    await Notification.applyTransformations(notifications);
-    return notifications.map((notification) => {
-      return new UserMenuNotificationItem({
-        notification,
-        currentUser: this.currentUser,
-        siteSettings: this.siteSettings,
-        site: this.site,
+
+    const content = [];
+    const data = await ajax("/notifications", { data: params });
+
+    const notifications = await Notification.initializeNotifications(
+      data.notifications
+    );
+
+    const reviewables = data.pending_reviewables?.map((r) =>
+      UserMenuReviewable.create(r)
+    );
+
+    if (reviewables?.length) {
+      const firstReadNotificationIndex = notifications.findIndex((n) => n.read);
+      const unreadNotifications = notifications.splice(
+        0,
+        firstReadNotificationIndex
+      );
+      mergeSortedLists(
+        unreadNotifications,
+        reviewables,
+        (notification, reviewable) => {
+          const notificationCreatedAt = new Date(notification.created_at);
+          const reviewableCreatedAt = new Date(reviewable.created_at);
+          return reviewableCreatedAt > notificationCreatedAt;
+        }
+      ).forEach((item) => {
+        if (item instanceof Notification) {
+          content.push(
+            new UserMenuNotificationItem({
+              notification: item,
+              currentUser: this.currentUser,
+              siteSettings: this.siteSettings,
+              site: this.site,
+            })
+          );
+        } else {
+          content.push(
+            new UserMenuReviewableItem({
+              reviewable: item,
+              currentUser: this.currentUser,
+              siteSettings: this.siteSettings,
+              site: this.site,
+            })
+          );
+        }
```

Comment:
```suggestion
        const props = {
          notification: item,
          currentUser: this.currentUser,
          siteSettings: this.siteSettings,
          site: this.site,          
        }

        const klass = (item instanceof Notification) ? UserMenuNotificationItem : UserMenuReviewableItem

        content.push(new klass(props))
```

Can you see if this is possible? We can write way less lines of code here.

Reply:
should be better now!